calico_test/tests/st/utils/docker_host.py

```diff
@@ -95,11 +96,29 @@ def calicoctl(self, command):
         :return: The output from the command with leading and trailing
         whitespace removed.
         """
+        etcd_ssl_type = os.environ.get("ETCD_SECURITY_TYPE", None)
         calicoctl = os.environ.get("CALICOCTL", "/code/dist/calicoctl")
+
         etcd_auth = "ETCD_AUTHORITY=%s:2379" % get_ip()
         # Export the environment, in case the command has multiple parts, e.g.
         # use of | or ;
         calicoctl = "export %s; %s" % (etcd_auth, calicoctl)
+        if etcd_ssl_type == "ssl":
+            if self.dind:
+                etcd_ca = ETCD_CA_FILE
+                etcd_cert = ETCD_CERT_FILE
+                etcd_key = ETCD_KEY_FILE
+            else:
+                cert_path = os.getenv("HOST_CERT_DIR")
```

Comment:
Could you not have passed a HOST_CERT_DIR through to the DinD container that allows you to just have a single branch of code?